src/model/models/modelManager.ts

```diff
@@ -61,4 +65,58 @@ export class ModelManager {
       throw new AppError('', httpStatus.INTERNAL_SERVER_ERROR, 'store-trigger service is not available', true);
     }
   }
+
+  public async deleteModel(identifier: string): Promise<StoreTriggerResponse> {
+    this.logger.debug({ msg: 'delete record', modelId: identifier });
+
+    let record: Record3D | undefined;
+
+    try {
+      const record: Record3D | undefined = await this.catalog.getRecord(identifier);
+
+      if (record === undefined) {
+        this.logger.error({ msg: 'model identifier not found', modelId: identifier });
+        throw new AppError('NOT_FOUND', httpStatus.NOT_FOUND, `Identifier ${identifier} wasn't found on DB`, true);
+      }
+
+      if (record.productStatus !== RecordStatus.UNPUBLISHED) {
+        this.logger.error({ msg: 'got UNPUBLISHED model', modelId: identifier });
+        throw new AppError(
+          'BAD_REQUEST',
+          httpStatus.BAD_REQUEST,
+          `Model ${record.productName} is PUBLISHED. The model must be UNPUBLISHED to be deleted!`,
+          true
+        );
+      }
+      this.logger.info({ msg: 'starting deleting record', modelId: identifier, modelName: record.productName });
+
+      const request: StoreTriggerDeletePayload = {
+        modelId: identifier,
+        pathToTileset: this.extractLink(record.links) as string,
+        modelName: record.productName,
+      };
+
+      const response: StoreTriggerResponse = await this.storeTrigger.deletePayload(request);
+      return response;
+    } catch (error) {
+      if (error instanceof AppError) {
+        throw error;
+      }
+
+      this.logger.error({ msg: 'Error in deleting record', identifier, modelName: record?.producerName, error, record });
```

Comment:
Remove the model name, it is not safe to use it because the record can be undefined. You already added the record itself so setting also the model name is unnecessary

Reply:
Remove the model name.

Reply:
removed